Jeewoogaram

GaMaius opened the pull request:
# 변경점 👍
week02까지의 과제

GaMaius committed:
feat(playerMove.cs, spawner.cs)

플레이어 움직임, 적 생성 및 플레이어 추적 스크립트 작성 및 적용

GaMaius committed:
feat(playerMove.cs)

플레이어 움직임에 따른 X축 플립 설정.

GaMaius committed:
feat(map)

맵 타일 그려서 배경 채움.

GaMaius committed:
fix(spawner.cs), feat(enemyAI.cs)

움직임 제어 코드 분리 및 코멘트 수정사항 변경.

GaMaius committed:
feat(weapon2), fix(enemyAI.cs)

플레이어 주위를 공전하는 무기를 생성,
데미지 입히는 코드를 arrow.cs에서 enemyAI.cs로 옮겨 여러 무기에 적용 가능하게 제작함.
오류 : 플레이어 진행방향에 따라 공전하는 무기의 속도가 달라짐. 차후 개선 예정.

GaMaius committed:
Arrow 발사 수정

arrow가 가까운 적에게로 날아가게 수정됨.

GaMaius committed:
LaserGun 추가

아직 레이저 발사는 안됨.
총이 캐릭터가 바라보는 시야 안의 적 중 가장 가까운 적에게 총구를 겨눔.

GaMaius committed:
feat(Laser)

Laser구현함.

GaMaius committed:
namespace, folder 설정

GaMaius committed:
Merge pull request #15 from ApptiveDev/Jeewoogaram_week02

Jeewoogaram week02 - Add Weapons

GaMaius committed:
Tile map Error 해결

GaMaius committed:
Merge branch 'Jeewoogaram_week02' into Jeewoogaram

GaMaius committed:
fix(con.unity.ide.vscode)

오류가 있는 라이브러리(?) 삭제

GaMaius committed:
feat(expBar, exp)

exp prefab 제작 및 애니메이션 적용
expBar는 Slider를 이용해 구현
exp는 처음에 5개, 그다음에는 레벨이 올라갈 때마다 획득해야 하는 경험치 양을 2배씩 증가시킴.

GaMaius committed:
fix(expBar, Player)

expBar가 게임 화면에서 안 보이는 버그를 해결
exp가 실제로 2배로 필요한 ManageExp에서의 버그 해결

hwanginseop commented on Assets/JWGR/Script/Player.cs:
```diff
@@ -0,0 +1,95 @@
+        // 게임이 작동하기 시작할 때 함수 실행.
+        private void Awake()
+        {
+            transform.position = Vector3.zero;
```
오 위치 초기화 시작하고 들어가는거 좋네요

hwanginseop commented on Assets/JWGR/Script/Player.cs:
```diff
@@ -0,0 +1,95 @@
+        }
+
+        // 물리 엔진 업데이트.
+        private void FixedUpdate()
```
오 fixedUpdate를 이용하여 고정된 성능 보장 좋습니다.

hwanginseop commented on Assets/JWGR/Script/Player.cs:
```diff
@@ -0,0 +1,95 @@
+                collisionObject = collision.gameObject;
+                if (collisionObject.gameObject.CompareTag("Enemy")) // 충돌한 상대가 적일 때
+                {
+                    health -= collisionObject.GetComponent<enemyInfo>().damage;
```
충돌한 오브젝트의 컴포넌트를 가져오는 참조방식 효율적인거 같네요

hwanginseop commented on Assets/JWGR/Script/aroundWeapon.cs:
```diff
@@ -0,0 +1,39 @@
+{
+    public class aroundWeapon : MonoBehaviour
+    {
+        public GameObject player;
```
player를 inspector에서 넣어서 참조시키는 방식 같은데 player스크립트에 전역변수를 선언했으니까 아얘 싱글톤 화 시켜서 참조하는 방식도 좋을 것 같습니다.

hwanginseop commented on Assets/JWGR/Script/arrow.cs:
```diff
@@ -0,0 +1,76 @@
+
+        private void Start()
+        {
+            GetComponent<SpriteRenderer>().sortingOrder = 7;
```
이걸 코드로 넣은 이유가 궁금하네요. 뭐 문제는 없습니다. 사실 전 엔진에서만 order를 조정해서 코드로 쓸 생각을 평소에 못하긴 했습니다.

hwanginseop commented on Assets/JWGR/Script/arrow.cs:
```diff
@@ -0,0 +1,76 @@
+        private void Start()
+        {
+            GetComponent<SpriteRenderer>().sortingOrder = 7;
+            player = GameObject.Find("Player");
```
오 이번엔 find네요. 다양한 방식으로 참조 할 수 있다는 점에서 좋지만 이왕이면 통일시키는게 좋을 듯 합니다.

hwanginseop commented on Assets/JWGR/Script/arrow.cs:
```diff
@@ -0,0 +1,76 @@
+            }
+
+            // 타겟 방향으로 이동
+            transform.position += moveDir * moveSpeed * Time.deltaTime;
```
화살이 날아가는 로직인거 같은데 화살의 이동, 거리 계산, 파괴는 코루틴과 while문을 활용하고 rotate는 필요한 시점에만 시켜서 좀 더 효율적으로 할 수 있을 듯 합니다.

hwanginseop commented on Assets/JWGR/Script/arrow.cs:
```diff
@@ -0,0 +1,76 @@
+            }
+        }
+
+        private Transform FindClosestEnemy()
```
오 거리제곱으로 루트연산 회피하는  방식 좋네요. 로직 잘짜셨네요.

hwanginseop commented on Assets/JWGR/Script/arrowSpawner.cs:
```diff
@@ -0,0 +1,27 @@
+            StartCoroutine(Spawn());
+        }
+
+        private IEnumerator Spawn()
```
b

hwanginseop commented on Assets/JWGR/Script/AutoLaser.cs:
```diff
@@ -0,0 +1,76 @@
+            {
+                laserRenderer.flipX = true;
+                FPRenderer.flipX = true;
+                LeanTween.scaleX(laserObj, -maxLength, drawDuration).setOnComplete(() => Destroy(laserObj));
```
오 leanTween을 활용한 레이저이펙트 시도 좋네요

hwanginseop commented on Assets/JWGR/Script/enemyAI.cs:
```diff
@@ -0,0 +1,82 @@
+            return direction;
+        }
+
+        private void OnTriggerEnter2D(Collider2D collision) // 충돌했을 때
```
Weapon2가 뭘까요?
그리고 exp instantiate할 때 처음엔 clone에 지정해서 생성하고 이후에는 그렇게 하지 않은게 딱히 이유가 없다면 통일하는게 좋아보입니다.

사실 근본적으로 무기에 맞으면 체력을 깎는다 라는 행위를 enemy에 달아 놓는 것 보다는 각 무기쪽에 enemy 체력을 깎도록 달거나 공통된 무기 class를 생성하여 거기에 구현하는게 좀 더 좋아보입니다.

hwanginseop commented on Assets/JWGR/Script/enemyInfo.cs:
오 enemy의 정보를 저장하는 일종의 구조체를 구현한 점 매우 좋습니다. 이후에 scriptableObj나 struct으로 구현하는 방식으로 발전 시켜도 좋을 것 같네요.

hwanginseop reviewed (commented):
과제 고생하셨습니다. 전체적으로 코드 및 여러 시도가 좋네요. 다만 무기 선택UI나 GameOver시 게임이 정지는 기능이 구현이 되지 않은게 아쉽습니다.

GaMaius committed:
feat(UI)

UI 구성
업그레이드 시스템 미구현 ㅜ

GaMaius committed:
feat(원거리 적, Upgrade System)

원거리 적을 추가하고
업그레이드 시스템을 제작함
오류가 발생했지만 원인을 못찾겠음

GaMaius committed:
fix(weaponManager, Editor Bug)

불필요한 스크립트 제거 및 버그 해결!

GaMaius committed:
feat(Boss, SoundManager, EventManager)

boss, sound, event 구현

GaMaius committed:
fix(sound, boss)

bgm과 sfx 정상출력되게 수정
boss 생성과 움직임 제대로 되게 수정
---
현재 무기가 다음 게임까지 레벨이 이어지며, 무기 활성화 비활성화가 이상하게 진행됨

GaMaius committed:
fix(Sound Bug)

보스 등장 후 BGM이 바뀔 때 발생한 소리 재생 오류 해결

GaMaius committed:
Bug 해결 실패

...

hwanginseop commented:
음 제가 잘못 플레이 한건지는 잘 모르겠는데 처음 무기 선택 후 무기가 안나옵니다. 그리고 코드 보니까 보스가 특수한 행동이 특별히 없는거 같은데 추가해주면 좋을것 같습니다. 외에는 다 깔끔하게 짜셨네요. 고민한 흔적이 많이 보입니다. 실력이 상당하다는 생각이 듭니다. 특히 데이터와 구현의 분리가 잘 이루어져있는 것이 매우 좋습니다.

GaMaius committed:
fix(Upgrade Bug)

Bug 해결 완료!(사실 다 못했어요)

GaMaius committed:
fix(Upgrade Bug, UI Bug)

무기 선택 시 생기는 오류 해결 및 Update에서 프리팹을 수정하며 매 실행마다 값이 초기화가 안되는 문제를 해결

GaMaius committed:
feat(PlayerAnimation)

플레이어가 서있을 때와 움직일 때 애니메이션 구현

GaMaius committed:
feat(HpBar)

HpBar구현 및 Manager들의 싱글톤 구현 / 씬이 로드될 때 파괴되지 않도록 설정

GaMaius committed:
feat(Money)

5f초마다 50G씩 늘어나는 Money를 추가
